spec/acceptance/reboot_finished_spec.rb

```diff
@@ -0,0 +1,39 @@
+require 'spec_helper_acceptance'
+
+describe 'Reboot Module - Reboot when Finished' do
+
+  def apply_reboot_manifest(agent, reboot_manifest)
+    apply_manifest_on agent, reboot_manifest do |result|
+      assert_match /defined 'message' as 'step_2'/,
+                   result.stdout, 'Expected step was not finished before reboot'
+    end
+    retry_shutdown_abort(agent)
+  end
+
+  reboot_manifest = <<-MANIFEST
```

Comment:
use rspec's `let(:reboot_manifest)` for this to avoid poisoning independent tests when (accidentally) changing this value.

See https://relishapp.com/rspec/rspec-core/v/3-7/docs/helper-methods/let-and-let for examples.